provider/manual/environ.go

```diff
@@ -239,26 +239,36 @@ func (e *manualEnviron) DestroyController(controllerUUID string) error {
 	script := `
 set -x
 touch %s
-# If jujud is running, we then wait for a while for it to stop.
+
 stopped=0
-if pkill -%d jujud; then
+function wait_for_jujud {
+    # Give jujud a chance to stop after pkill.
     for i in {1..30}; do
         if pgrep jujud > /dev/null ; then
             sleep 1
         else
-            echo "jujud stopped"
+            echo jujud stopped
             stopped=1
             logger --id jujud stopped on attempt $i
             break
         fi
     done
-fi
-if [ $stopped -ne 1 ]; then
+}
+
+# There might be no jujud at all (for example, after a failed deployment) so
+# don't require pkill to succeed before checking jujud is dead.
+pkill -%d jujud
+wait_for_jujud
+
+[[ $stopped -ne 1 ]] && {
     # If jujud didn't stop nicely, we kill it hard here.
-    %spkill -9 jujud
-    service %s stop
-    logger --id killed jujud and stopped %s
-fi
+    %spkill -9 jujud && wait_for_jujud
+}
+[[ $stopped -ne 1 ]] && {
+    echo jujud removal failed
+    exit 1
+}
+service %s stop && logger --id stopped %s
```

Comment:
Figuring out what all of these %s actually map to is pretty hard. Looking online I see Nate got a response of:
https://groups.google.com/forum/#!topic/golang-nuts/4FbzMw7WMck

Use "text/template" instead of fmt.Sprintf()
Is it hard to change that here? If it isn't, it would certainly make the code more readable.